### PR TITLE
source-postgres: connect with adaptiveFetch=true, take 2

### PR DESCRIPTION
**Pull request by postamar**
## What
Fixes https://github.com/airbytehq/airbyte-internal-issues/issues/7767
See https://github.com/airbytehq/airbyte/pull/38365
See https://github.com/airbytehq/airbyte/pull/38289

## How
Disable autocommit and connect with extra JDBC URL parameters

edit: this was more involved than I thought it would be. Autocommit is being turned off correctly, unlike what I thought (it's turned off after the statement is created, but in postgres, that's OK). Adaptive fetch is also active, as I found out by stepping through the debugger. The problem turns out to be that the adaptive fetch implementation is not very good:
1. it requires setting a hard limit in the form of `maxResultBuffer` which throws an exception immediately when hit
2. adaptive fetch tries to fill up the ResultSet as much to capacity as possible
3. its prediction is just a prediction and sometimes it's wrong and which causes the exception in (1) to be thrown.

I worked around this by setting a hard limit of 1000 rows in an effort to mitigate these exceptions. If this doesn't work out, we need to disable adaptive fetching again and simply set the fetch size to some constant value of 10 or something. It's basically what sqlserver and mysql do. Hopefully the impact on performance won't be too bad but who knows.

## User Impact
This shouldn't cause any perceptible changes, other than OOMs being eliminated, or, at least, occuring much more rarely.

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java [hunk 1]
```diff
@@ -150,6 +156,22 @@ public class PostgresSource extends AbstractJdbcSource<PostgresType> implements
   public static final String SSL_MODE_DISABLE = "disable";
   public static final String SSL_MODE_REQUIRE = "require";
 
+  public static final Map<PGProperty, String> JDBC_CONNECTION_PARAMS = ImmutableMap.of(
+      // Initialize parameters with prepareThreshold=0 to mitigate pgbouncer errors
+      // https://github.com/airbytehq/airbyte/issues/24796
+      PREPARE_THRESHOLD, "0",
+      DEFAULT_ROW_FETCH_SIZE, "1",
+      ADAPTIVE_FETCH, "true",
+      MAX_RESULT_BUFFER, "10percent",
```
Comment (rodireich) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java:165]: this sets a limit of 10% of RAM / up to 1000 rows pet fetch?
Wouldn't that slow us down? we typically fetch rows numbered in 100Ks if not millions.

**Review by rodireich — APPROVED** [review]
Change look good. I had a question regarding the limit we put on fetching.

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java (hunk 1 above)
Comment (postamar) [reply airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java:165]: I wondered about that. Mysql fetches 1 row at a time, and SQL Server fetches 10 rows. Perhaps we can run a performance test. Let's discuss on Monday, in the meantime we should not merge this on a Friday.

Comment (rodireich) [reply airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java:165]: > Mysql fetches 1 row at a time, and SQL Server fetches 10 rows 
I don't think that's the case. we start by taking a sample of 10. But AFAIK current fetching limit is attempting to fill up to 50% of RAM

Comment (theyueli) [reply airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java:165]: do we have any performance benchmark we can use as a quick regression test?

Comment (akashkulk) [reply airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java:165]: Curious here : why the significant departure from our default fetch settings? 
https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/streaming/FetchSizeConstants.kt

Comment (postamar) [reply airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java:165]: The comments I committed should explain this.

It's very possible that this adaptiveFetch feature just plain sucks and is inferior to our own. Lets see if the postgres performance takes a hit and worst case we revert this PR.

**Comment by postamar** [comment]
I poked around in metabase since merging this yesterday and it looks like avg throughput remains unaffected by this change. It remains to be seen whether OOMs have been mitigated, of course.

**Pull request merged**